### ISSUE TITLE
The list of referenced files is not displayed correctly

### ISSUE THREAD

**Issue by zx616456607** [clangd/clangd#1947]
The reference file of the function has been renamed or deleted, and when you view the reference again, the reference list file shows a file that does not exist

New header file testh.h
Contents of the file:
void swap(int a, int b);

New source file testc.cpp
Contents of the file:
#include "testh.h"

void swap (int a, int b) {
  Int temp = a
  A = b
  B = temp
}

New source file testmain.cpp
Contents of the file:
#include "testh.h"

int main() {
    int a = 100;
    int b = 200;
    swap(a, b);
    return 0;
}

In the testc.cpp file, right-click to go to the reference to view the list of swap function references.
![image](https://github.com/clangd/clangd/assets/19658190/09b6bd52-a425-4750-8a60-3147816231d6)

Rename testmain.cpp to testmain1.cpp. In the testc.cpp file, right-click to go to reference to view the list of swap function references.
![image](https://github.com/clangd/clangd/assets/19658190/8d04551b-249b-4a28-b5ad-61e685512a1f)

Delete the testmain1.cpp file. In the testc.cpp file, right-click to go to the reference to view the list of swap function references.
![image](https://github.com/clangd/clangd/assets/19658190/0fa98008-692d-4382-8aea-fa991f85d6b9)

**Logs**

Please attach the clangd stderr log if you can. (Usually available from the editor)
If possible, run with `--log=verbose` - note that the logs will include the contents of open files!
If this is a crash, try to put `llvm-symbolizer` on your PATH per the troubleshooting instructions.

**System information**

Output of `clangd --version`: 16.0.0

Editor/LSP plugin:  vscode-clangd 0.1.24

Operating system: Windows 10

**Comment by HighCommander4** [comment]
Does restarting clangd (or restarting your editor) after renaming/deleting the file fix the issue?

**Comment by zx616456607** [comment]
> Does restarting clangd (or restarting your editor) after renaming/deleting the file fix the issue?

Restarting clangd or restarting the editor can solve the problem, Is this a bug?

**Comment by HighCommander4** [comment]
> Is this a bug?

Yes, I think this is still a valid bug. Ideally, clangd should update its index on the fly to reflect file renames and deletions.

(If the issue persisted across a restart, that would have been a more severe bug.)

**Comment by HighCommander4** [comment]
See also #1104 for another situation where clangd does not update its index properly while running, but only on restart.

**Comment by ckandeler** [comment]
Also [#935](https://github.com/clangd/clangd/issues/935).

**Comment by youssef-lr** [comment]
Are there plans to fix this or is a PR to do so welcome?

**Comment by HighCommander4** [comment]
> Are there plans to fix this or is a PR to do so welcome?

I expect a PR to fix this would be welcome.
